coremltools/models/model.py

```diff
@@ -602,3 +615,40 @@ def _verify_input_name_exists(self, input_dict):
                 err_msg = "Provided key \"{}\", in the input dict, " \
                           "does not match any of the model input name(s), which are: {}"
                 raise KeyError(err_msg.format(given_input, ",".join(model_input_names)))
+
+    _supported_dtype = [
+        numpy.float32,
+        numpy.float64,
+        numpy.int32,
+    ]
+
+    def _verify_input_type(self, input_dict):
+        model_input_types = {}
+        for inp in self._spec.description.input:
+            type_value = inp.type.multiArrayType.dataType
+            type_name = inp.type.multiArrayType.ArrayDataType.Name(type_value)
+            if type_name != "INVALID_ARRAY_DATA_TYPE":
+                model_input_types[inp.name] = type_name
+
+        for given_input_name, given_input in input_dict.items():
+            if not given_input_name in model_input_types:
+                continue
+            input_dict[given_input_name] = self._convert_to_numpy_if_enable(given_input, given_input_name)
+            given_input = input_dict[given_input_name]
+            if given_input.dtype not in self._supported_dtype:
+                err_msg = "dtype of {} is {}, it is not supported"
+                raise ValueError(err_msg.format(given_input_name, given_input.dtype))
+
+    def _convert_to_numpy_if_enable(self, given_input, given_input_name):
+        if isinstance(given_input, numpy.ndarray):
+            sanitized_input = given_input
+        elif _HAS_TORCH and isinstance(given_input, torch.Tensor):
+            sanitized_input = given_input.detach().numpy()
+        elif (_HAS_TF_1 or _HAS_TF_2) and isinstance(given_input, tf.Tensor):
+            sanitized_input = given_input.eval(session=tf.compat.v1.Session())
+        else:
+            sanitized_input = numpy.array(given_input)
```

Comment:
`numpy.array` will throw an exception if it can't do the conversions. So I think this code should be:
```python
try: 
    sanitized_input = numpy.array(given_input)
except:
    err_msg = "Can not convert {} parameter to a numpy array."
    raise ValueError(err_msg.format(given_input_name))
```

Reply:
Ok, then let's just remove the if-statement in this else-block. That's not ideal, but this will still be a large improvement over the current functionality in `main`.